sdk/python/kfp_tekton/compiler/compiler.py

```diff
@@ -482,31 +424,79 @@ def _create_pipeline_workflow(self, args, pipeline, op_transformers=None, pipeli
         include_loop_task_refs.extend(with_loop_task)
       task_refs = include_loop_task_refs
 
-    # generate the Tekton Pipeline document
-    pipeline_template = {
+    # TODO: generate the PipelineRun template
+    pipeline_run = {
       'apiVersion': tekton_api_version,
-      'kind': 'Pipeline',
+      'kind': 'PipelineRun',
       'metadata': {
-        'name': pipeline.name or 'Pipeline',
-        'annotations': {
+        'name': sanitize_k8s_name(pipeline.name or 'Pipeline', suffix_space=4) + '-run',
+        'annotations': {  # TODO: do annotations apply to PipelineRun?
           # Disable Istio inject since Tekton cannot run with Istio sidecar
           'sidecar.istio.io/inject': 'false'
         }
       },
       'spec': {
-        'params': params,
-        'tasks': task_refs
+        'params': [{
+            'name': p['name'],
+            'value': p.get('default', '')
+          } for p in params],
+        'pipelineSpec': {
+          'params': params,
+          'tasks': task_refs
+        }
       }
     }
 
-    # append Task and Pipeline documents
-    workflow = templates + [pipeline_template]
+    # TODO: pipelineRun additions
+
+    # Generate TaskRunSpec PodTemplate:s
+    task_run_spec = []
+    for task in task_refs:
+      # TODO: should loop-item tasks be included here?
+      if LoopArguments.LOOP_ITEM_NAME_BASE in task['name']:
+        task_name = re.sub(r'-%s-.+$' % LoopArguments.LOOP_ITEM_NAME_BASE, '', task['name'])
+      else:
+        task_name = task['name']
+      op = pipeline.ops.get(task_name)
+      if not op:
+        raise RuntimeError("unable to find op with name '%s'" % task["name"])
+      task_spec = {"pipelineTaskName": task['name'],
+                   "taskPodTemplate": {}}
+      if op.affinity:
+        task_spec["taskPodTemplate"]["affinity"] = convert_k8s_obj_to_json(op.affinity)
+      if op.tolerations:
+        task_spec["taskPodTemplate"]['tolerations'] = op.tolerations
+      if op.node_selector:
+        task_spec["taskPodTemplate"]['nodeSelector'] = op.node_selector
+      if bool(task_spec["taskPodTemplate"]):
+        task_run_spec.append(task_spec)
+    if len(task_run_spec) > 0:
+      pipeline_run['spec']['taskRunSpec'] = task_run_spec
+
+    # add workflow level timeout to pipeline run
+    if pipeline.conf.timeout:
+      pipeline_run['spec']['timeout'] = '%ds' % pipeline.conf.timeout
+
+    # generate the Tekton service account template for image pull secret
+    service_template = {}
+    if len(pipeline.conf.image_pull_secrets) > 0:
+      service_template = {
+        'apiVersion': 'v1',
+        'kind': 'ServiceAccount',
+        'metadata': {'name': sanitize_k8s_name(pipeline_run['metadata']['name'], suffix_space=3) + '-sa'}
+      }
+    for image_pull_secret in pipeline.conf.image_pull_secrets:
+      service_template['imagePullSecrets'] = [{'name': image_pull_secret.name}]
+
+    # TODO: service_template
```

Comment:
Today the image secret pull PR is finally merged in Tekton, you can ignore this for now and I can create a separate PR to fix it with the new implementation.